packages/vite/src/node/config.ts

```diff
@@ -330,17 +337,19 @@ export async function resolveConfig(
   configEnv.mode = mode
 
   // resolve plugins
-  const rawUserPlugins = (config.plugins || []).flat(Infinity).filter((p) => {
-    if (!p) {
-      return false
-    } else if (!p.apply) {
-      return true
-    } else if (typeof p.apply === 'function') {
-      return p.apply({ ...config, mode }, configEnv)
-    } else {
-      return p.apply === command
+  const rawUserPlugins = (await asyncFlatten(config.plugins || [])).filter(
+    (p: any) => {
```

Comment:
Given the type predicate above, `p`'s type is properly inferred as `PluginOption` here.

```suggestion
    (p) => {
```

Reply:
LGTM